core/modules/memman/MemFile.cc

```diff
@@ -0,0 +1,179 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "memman/MemFile.h"
+
+// System Headers
+#include <errno.h>
+#include <mutex>
+#include <unordered_map>
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+/******************************************************************************/
+/*                  L o c a l   S t a t i c   O b j e c t s                   */
+/******************************************************************************/
+  
+namespace {
+std::mutex                                 cacheMutex;
+std::unordered_map<std::string, MemFile *> fileCache;
+}
```

Comment:
I dislike local static objects, but I'm having trouble coming up with a legitimate complaint. I'd feel better if these were members of MemMan or some other managing class. They are pretty well buried here and it limits it to one instance of MemManReal, but I can't think of any case where there would be a need for more than one MemMan.